Epoching:   0%|                                                                                                                                                                                                                                                                                 | 0/5 [00:00<?, ?it/s]Passing a tuple of `past_key_values` is deprecated and will be removed in Transformers v4.48.0. You should pass an instance of `EncoderDecoderCache` instead, e.g. `past_key_values=EncoderDecoderCache.from_legacy_cache(past_key_values)`.
  0%|â–Œ                                                                                                                                                                                                                                                                                | 1/472 [00:01<08:35,  1.09s/it]
Epoching:   0%|                                                                                                                                                                                                                                                                                 | 0/5 [00:01<?, ?it/s]
Traceback (most recent call last):
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 55, in <module>
    loss = wrapper(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/Desktop/learning-nlp/_modules.py", line 19, in forward
    outputs = self.model(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/transformers/models/t5/modeling_t5.py", line 1941, in forward
    loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/loss.py", line 1297, in forward
    return F.cross_entropy(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/functional.py", line 3494, in cross_entropy
    return torch._C._nn.cross_entropy_loss(
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 0 has a total capacity of 11.90 GiB of which 1.71 GiB is free. Including non-PyTorch memory, this process has 10.18 GiB memory in use. Of the allocated memory 8.18 GiB is allocated by PyTorch, and 1.82 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
Traceback (most recent call last):
  File "/home/melaichouni/Desktop/learning-nlp/asho_project.py", line 55, in <module>
    loss = wrapper(input_ids=input_ids, attention_mask=attention_mask, labels=labels)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/Desktop/learning-nlp/_modules.py", line 19, in forward
    outputs = self.model(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/transformers/models/t5/modeling_t5.py", line 1941, in forward
    loss = loss_fct(lm_logits.view(-1, lm_logits.size(-1)), labels.view(-1))
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1751, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1762, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/modules/loss.py", line 1297, in forward
    return F.cross_entropy(
  File "/home/melaichouni/.virtualenvs/dev/lib/python3.10/site-packages/torch/nn/functional.py", line 3494, in cross_entropy
    return torch._C._nn.cross_entropy_loss(
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1.91 GiB. GPU 0 has a total capacity of 11.90 GiB of which 1.71 GiB is free. Including non-PyTorch memory, this process has 10.18 GiB memory in use. Of the allocated memory 8.18 GiB is allocated by PyTorch, and 1.82 GiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
